example-playbooks/restore_scylla_manager_backup/restore.yaml

```diff
@@ -0,0 +1,135 @@
+        name: scylla-server
+        state: stopped
+
+    - name: Remove Scylla directories
+      file:
+        path: "/var/lib/scylla/{{ item }}"
+        state: absent
+      loop:
+        - data
+        - commitlog
+        - hints
+        - view_hints
+
+    - name: Recreate Scylla directories
+      file:
+        path: "/var/lib/scylla/{{ item }}"
+        state: directory
+        owner: scylla
+        group: scylla
+        mode: 0755
+      loop:
+        - data
+        - commitlog
+        - hints
+        - view_hints
+
+    - name: Get seeds list
+      shell: |
+        grep '\- seeds:' /etc/scylla/scylla.yaml | awk '{ print $NF }' | tr -d '"' | tr -d "'"
+      register: _seeds
+
+    - name: Get initial token
+      shell: |
+        scylla-manager-agent download-files -L {{ backup_location }} -n {{ host_id[inventory_hostname] }} -T {{ snapshot_tag }} --dump-tokens
+      register: _initial_token
+
+    - name: Set initial_token in scylla.yaml
+      lineinfile:
+        path: /etc/scylla/scylla.yaml
+        regexp: '^(#\s*)?initial_token:'
+        line: "initial_token: {{ _initial_token.stdout }}"
+        create: yes
+
+    - name: Make each node a seed
+      lineinfile:
+        path: /etc/scylla/scylla.yaml
+        regexp: '^(\s+) - seeds:'
+        line: |
+          \g<1> - seeds: {{ groups.all | join(',') }}
+        backrefs: yes
+
+    - name: Start Scylla service
+      service:
+        name: scylla-server
+        state: started
+
+    - name: Wait for CQL port
+      wait_for:
+        port: 9042
+
+    - name: Stop Scylla service
+      service:
+        name: scylla-server
+        state: stopped
+
+- name: Restore schema
+  hosts: all[0]
+  become: true
+  gather_facts: false
+  tasks:
+    - name: Remove other seeds
+      lineinfile:
+        path: /etc/scylla/scylla.yaml
+        regexp: '^(\s+) - seeds:'
+        line: |
+          \g<1> - seeds: {{ inventory_hostname }}
+        backrefs: yes
+
+    - name: Download system_schema
+      shell: |
+        scylla-manager-agent download-files -L {{ backup_location }} -n {{ host_id[inventory_hostname] }} -T {{ snapshot_tag }} -d /var/lib/scylla/data/ -K system_schema --silent
+      become_user: scylla
+
+    - name: Start Scylla service
+      service:
+        name: scylla-server
+        state: started
+
+- name: Restore data
+  hosts: all
+  become: true
+  gather_facts: false
+  tasks:
+    - name: Start Scylla service
+      service:
+        name: scylla-server
+        state: started
+
+    - name: Wait for CQL port
+      wait_for:
+        port: 9042
+
+    - name: Stop Scylla service
+      service:
+        name: scylla-server
+        state: stopped
+
+    - name: Download data
+      shell: |
+        scylla-manager-agent download-files -L {{ backup_location }} -n {{ host_id[inventory_hostname] }} -T {{ snapshot_tag }} -d /var/lib/scylla/data/ -K '*,!system_schema' --silent
+      become_user: scylla
+      async: 86400
+      poll: 5
```

Comment:
What if it takes longer? This is not the best way to run an async task.
See here for a more robust way: https://docs.ansible.com/ansible/latest/collections/ansible/builtin/async_status_module.html#async-status-module

Reply:
After reading [the docs](https://docs.ansible.com/ansible/2.9/user_guide/playbooks_async.html) it seems that what you suggest is intended to join on a task you spawned before 

> If you would like to perform a task asynchronously and check on it later you can perform a task similar to the following:

```
---
# Requires ansible 1.8+
- name: 'YUM - async task'
  yum:
    name: docker-io
    state: present
  async: 1000
  poll: 0
  register: yum_sleeper

- name: 'YUM - check on async task'
  async_status:
    jid: "{{ yum_sleeper.ansible_job_id }}"
  register: job_result
  until: job_result.finished
  retries: 30
```

Here we want to block on the download, I sure can change the t/o from 1 day to 1 week.